<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AdjustBMParam</class>
 <widget class="QDialog" name="AdjustBMParam">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>551</width>
    <height>365</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BM Parameters</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Pre filter size</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QSlider" name="preFilterSz">
       <property name="minimum">
        <number>5</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="value">
        <number>21</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QLineEdit" name="pSz">
       <property name="text">
        <string>21</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Pre filter cap</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QSlider" name="preFilterCap">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>31</number>
       </property>
       <property name="value">
        <number>30</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QLineEdit" name="pCap">
       <property name="text">
        <string>30</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>SAD wndow size</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QSlider" name="SADWndSz">
       <property name="minimum">
        <number>5</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="singleStep">
        <number>2</number>
       </property>
       <property name="value">
        <number>21</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QLineEdit" name="sadSz">
       <property name="text">
        <string>21</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Texture threshold</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QSlider" name="textureth">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>200</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="3" column="2">
      <widget class="QLineEdit" name="textTh">
       <property name="text">
        <string>1</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Min. disparity</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QSlider" name="minDisp">
       <property name="minimum">
        <number>-30</number>
       </property>
       <property name="maximum">
        <number>130</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="4" column="2">
      <widget class="QLineEdit" name="minDisp_2">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Num.disparities</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QSlider" name="numDisp">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>20</number>
       </property>
       <property name="singleStep">
        <number>16</number>
       </property>
       <property name="pageStep">
        <number>16</number>
       </property>
       <property name="value">
        <number>6</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::NoTicks</enum>
       </property>
       <property name="tickInterval">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item row="5" column="2">
      <widget class="QLineEdit" name="numDisp_2">
       <property name="text">
        <string>128</string>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>Uniquness ratio</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QSlider" name="uniq">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>200</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="6" column="2">
      <widget class="QLineEdit" name="uniqRatio">
       <property name="text">
        <string>1</string>
       </property>
      </widget>
     </item>
     <item row="7" column="0">
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>Speckle win size</string>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <widget class="QSlider" name="speckleWndSz">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="7" column="2">
      <widget class="QLineEdit" name="spkSz">
       <property name="text">
        <string>1</string>
       </property>
      </widget>
     </item>
     <item row="8" column="0">
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>Speckle range</string>
       </property>
      </widget>
     </item>
     <item row="8" column="1">
      <widget class="QSlider" name="specklerange">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="8" column="2">
      <widget class="QLineEdit" name="spkRag">
       <property name="text">
        <string>1</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>378</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="tryBtn">
       <property name="text">
        <string>Save paramters</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>preFilterSz</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>updateFilterSize()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>18</y>
    </hint>
    <hint type="destinationlabel">
     <x>82</x>
     <y>6</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>preFilterCap</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>updateFilterCap()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>49</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SADWndSz</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>updateSADWinSz()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>98</y>
    </hint>
    <hint type="destinationlabel">
     <x>54</x>
     <y>57</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>textureth</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>updateTexture()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>82</x>
     <y>109</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>minDisp</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>updateMinDisp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>164</y>
    </hint>
    <hint type="destinationlabel">
     <x>2</x>
     <y>143</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>numDisp</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>updateMaxDisp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>175</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uniq</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>updtaeUniq()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>230</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>225</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>speckleWndSz</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>updateSpeckSz()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>263</y>
    </hint>
    <hint type="destinationlabel">
     <x>116</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>specklerange</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>updateSpeckRange()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>296</y>
    </hint>
    <hint type="destinationlabel">
     <x>163</x>
     <y>290</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>preFilterSz</sender>
   <signal>sliderReleased()</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>tryParam()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>22</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>5</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>preFilterCap</sender>
   <signal>sliderReleased()</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>tryParam()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>32</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SADWndSz</sender>
   <signal>sliderReleased()</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>tryParam()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>98</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>60</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>textureth</sender>
   <signal>sliderReleased()</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>tryParam()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>82</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>minDisp</sender>
   <signal>sliderReleased()</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>tryParam()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>164</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>109</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>numDisp</sender>
   <signal>sliderReleased()</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>tryParam()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uniq</sender>
   <signal>sliderReleased()</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>tryParam()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>230</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>164</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>speckleWndSz</sender>
   <signal>sliderReleased()</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>tryParam()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>263</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>189</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>specklerange</sender>
   <signal>sliderReleased()</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>tryParam()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>296</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tryBtn</sender>
   <signal>clicked()</signal>
   <receiver>AdjustBMParam</receiver>
   <slot>saveParameters()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>418</x>
     <y>346</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>321</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>tryParam()</slot>
  <slot>updateFilterSize()</slot>
  <slot>updateFilterCap()</slot>
  <slot>updateSADWinSz()</slot>
  <slot>updateTexture()</slot>
  <slot>updateSpeckSz()</slot>
  <slot>updateSpeckRange()</slot>
  <slot>updtaeUniq()</slot>
  <slot>updateMinDisp()</slot>
  <slot>updateMaxDisp()</slot>
  <slot>saveParameters()</slot>
 </slots>
</ui>
